extensions/podman/src/extension.ts

```diff
@@ -884,6 +903,58 @@ export function registerOnboardingMachineExistsCommand(): extensionApi.Disposabl
   });
 }
 
+export function registerOnboardingUnsupportedPodmanMachineCommand(): extensionApi.Disposable {
+  return extensionApi.commands.registerCommand('podman.onboarding.checkUnsupportedPodmanMachine', async () => {
+    let isUnsupported = false;
+    const installedPodman = await getPodmanInstallation();
+    if (installedPodman) {
+      isUnsupported = await shouldNotifyQemuMachinesWithV5(installedPodman);
+    }
+
+    extensionApi.context.setValue('unsupportedPodmanMachine', isUnsupported, 'onboarding');
+  });
+}
+
+export function registerOnboardingRemoveUnsupportedMachinesCommand(): extensionApi.Disposable {
+  return extensionApi.commands.registerCommand('podman.onboarding.removeUnsupportedMachines', async () => {
+    // only on macOS
+    // do not check if version is v5 as it is being checked by the command that triggers this one
+    if (extensionApi.env.isMac) {
+      // remove the qemu machines folder
+      const qemuSharePath = path.resolve(os.homedir(), appHomeDir(), 'machine', 'qemu');
+      const qemuConfigPath = path.resolve(os.homedir(), appConfigDir(), 'machine', 'qemu');
+
+      // remove folders if exists
+      const foldersToRemove = [];
+      if (fs.existsSync(qemuSharePath)) {
+        foldersToRemove.push(qemuSharePath);
+      }
+      if (fs.existsSync(qemuConfigPath)) {
+        foldersToRemove.push(qemuConfigPath);
+      }
+
```

Comment:
A little redundant and maybe paranoid, but maybe an "if (foldersToRemove.isEmpty) return" here? e.g. maybe there is a delay and user has deleted them since the check.

Reply:
so the check here is done during the onboarding flow, I mean, it's called if user click on a button so I think there is no time to click and run a command elsewhere

Reply:
ok 👍🏼